src/main/java/com/files/AddressValidator.java

```diff
@@ -0,0 +1,47 @@
+package com.files;
+
+import com.log.Logs;
+import com.setup.Setup;
+
+import java.net.URL;
+import java.util.logging.Logger;
+
+public class AddressValidator {
```

Comment:
@PavelNejinskiy the concept of `AddressValidator` and `UrlValidator` can fit into different `Images` implementation. I'd name it `ValidateImages`. This implementation should expect some origin `Images` implementation at constructor and `Extensions` instance. As for its `add` method - it is the place where the validation should happen. In case extension is not supported, lets throw our own exception(lets create one) - `ExtensionNotSupported`, lets derive it from `RuntimeException`